### PR TITLE
multiple fixes + tests

### PR DESCRIPTION
**Pull request by dime-squanderer**

### CODE REVIEW

File: integration_test/import_address_pk_test.dart [hunk 1]
```diff
@@ -33,40 +34,40 @@ void main() {
...
-              "cSTCxFTQ6YuvmiXS8ETMB4r1r2WyWs2cdYxvPGWSXcUYahvZ2ki8",
-          'mkUtf98ZFvo59ewRvH86yrmFxcXtfG2K1p':
-              "cNKSrFXwQJrbCgTJabkDutEoSH3CKGHnbomGUNobrhfRCCjkiyL1",
-          'tb1qxe6v9m4ekyuxljh9cl86yarjutxjw02kleka6a':
-              "cNKSrFXwQJrbCgTJabkDutEoSH3CKGHnbomGUNobrhfRCCjkiyL1",
-          'mnfimZCF7gaR6qhzDHnB9NKFDfdYM9QXDN':
-              "cUAgAnRc6D9uVVLGAtiHUosHQjc8Fwbo4CbXLxgBhxbK1krdAHm7",
-          'tb1qfec424jsfhawu4cw7353p6qu48yc46924rujet':
-              "cUAgAnRc6D9uVVLGAtiHUosHQjc8Fwbo4CbXLxgBhxbK1krdAHm7",
+          // 'tb1q8dgyqdnyr6kf3ctawy5ljl73r86h95aqwg8k7c':
```
Comment (dime-squanderer) [inline integration_test/import_address_pk_test.dart:37]: unnecessary to test all these -- takes forever to run. testing a few of each type should be sufficient

File: lib/data/sources/network/api/v2_api.dart [hunk 2]
```diff
@@ -4554,95 +4604,33 @@ abstract class V2Api {
...
-    @Query("asset") String? asset,
-    @Query("quantity") int? quantity,
     @Query("skip_validation") bool? skipValidation,
     @Query("allow_unconfirmed_inputs") bool? allowUnconfirmedInputs,
     @Query("exact_fee") int? exactFee,
     @Query("inputs_set") String? inputsSet,
+    @Query("exclude_utxos_with_balances") bool? excludeUtxosWithBalances,
     @Query("unconfirmed") bool? unconfirmed,
   ]);
-
```
Comment (dime-squanderer) [inline lib/data/sources/network/api/v2_api.dart:4585]: I removed all the commented docs since it made searching through the file difficult, and a lot of the docs have changed since this comment was created. I can add it back if you find it useful, I don't have a strong opinion either way.

File: integration_test/encryption_test.dart [hunk 3]
```diff
@@ -1,177 +1,202 @@
...
-      const password = 'testPassword123';
-
-      final encrypted =
-          await fallbackEncryptionService.encrypt(originalData, password);
-      expect(encrypted.contains('m=65536,t=6,p=4'), true);
-      await tester.pumpAndSettle();
-    });
-  });
-}
+// import 'package:flutter_test/flutter_test.dart';
```
Comment (mattmarcello) [inline integration_test/encryption_test.dart:1]: We should be able to run these, no?  Just swap in the EncryptionServiceWebWorker impl.

File: integration_test/import_address_pk_test.dart (hunk 1 above)
Comment (mattmarcello) [reply integration_test/import_address_pk_test.dart:37]: okay cool.

File: lib/data/models/transaction_info.dart [hunk 4]
```diff
@@ -7,7 +7,7 @@ class InfoVerboseMapper {
     return switch (info) {
       api.EnhancedSendInfoVerbose(unpackedData: var u) =>
         TransactionInfoEnhancedSend(
-          btcAmountNormalized: info.btcAmountNormalized,
+          btcAmountNormalized: info.btcAmountNormalized!,
```
Comment (mattmarcello) [inline lib/data/models/transaction_info.dart:10]: Why is this necessary?  

In general it's probably best to use the "??" with a default value instead of the `!` which could possibly throw.

File: lib/data/models/transaction_unpacked.dart [hunk 5]
```diff
@@ -135,3 +141,26 @@ class CancelUnpackedVerboseMapper {
...
+
+class DetachUnpackedVerboseMapper {
+  static DetachUnpackedVerbose toDomain(api.DetachUnpackedVerbose u) {
+    return DetachUnpackedVerbose(destination: u.destination);
+  }
+}
+
+class MoveToUtxoUnpackedVerboseMapper {
+  static MoveToUtxoUnpackedVerbose toDomain(api.MoveToUtxoUnpackedVerbose u) {
+    return const MoveToUtxoUnpackedVerbose();
```
Comment (mattmarcello) [inline lib/data/models/transaction_unpacked.dart:164]: this is just an empty class?

File: lib/domain/entities/transaction_unpacked.dart [hunk 6]
```diff
@@ -261,3 +261,26 @@ class CancelUnpacked extends TransactionUnpacked {
...
+}
+
+class DetachUnpackedVerbose extends TransactionUnpacked {
+  final String destination;
+  const DetachUnpackedVerbose({required this.destination})
+      : super(messageType: "detach");
+}
+
+class MoveToUtxoUnpackedVerbose extends TransactionUnpacked {
+  const MoveToUtxoUnpackedVerbose() : super(messageType: "move_to_utxo");
```
Comment (mattmarcello) [inline lib/domain/entities/transaction_unpacked.dart:285]: should this be "utxo_move"?  not sure.

File: lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart [hunk 7]
```diff
@@ -43,9 +44,24 @@ class FetchComposeAttachUtxoFormDataUseCase {
...
+                  assetName &&
+              balance.address == address &&
+              balance.utxo == null)
+          .toList();
+      if (balanceForAddress.isEmpty) {
+        throw FetchBalanceException('Balance not found for address: $address');
+      }
+      if (balanceForAddress.length > 1) {
+        throw FetchBalanceException(
+            'Multiple balances found for address: $address');
```
Comment (mattmarcello) [inline lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart:62]: Does this happen?  If so, is the correct behavior to throw an exception?

File: lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart [hunk 8]
```diff
@@ -43,9 +44,24 @@ class FetchComposeAttachUtxoFormDataUseCase {
...
-          .getBalancesForAddressAndAssetVerbose(address, assetName)
-          .then((balances) => balances.first);
+      final balances = await balanceRepository
+          .getBalancesForAddressAndAssetVerbose(address, assetName);
+      final balanceForAddress = balances
+          .where((balance) =>
+              displayAssetName(
+                      balance.asset, balance.assetInfo.assetLongname) ==
+                  assetName &&
+              balance.address == address &&
```
Comment (mattmarcello) [inline lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart:54]: is this check necessary?  doesn't the query imply that we are filtering by `address`?

File: lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart [hunk 9]
```diff
@@ -43,9 +44,24 @@ class FetchComposeAttachUtxoFormDataUseCase {
...
+              balance.utxo == null)
```
Comment (mattmarcello) [inline lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart:55]: We don't ever want to show balances that are attached to utxos?

Comment (mattmarcello) [reply lib/presentation/screens/compose_attach_utxo/usecase/fetch_form_data.dart:55]: How are balances with `utxo != null` handled in the application?

File: lib/presentation/screens/compose_detach_utxo/bloc/compose_detach_utxo_bloc.dart [hunk 10]
```diff
@@ -68,20 +55,21 @@ class ComposeDetachUtxoBloc extends ComposeBaseBloc<ComposeDetachUtxoState> {
     ));
 
     try {
-      final (feeEstimates, balance) =
-          await fetchComposeDetachUtxoFormDataUseCase
-              .call(event.currentAddress!);
+      final feeEstimates =
+          await getFeeEstimatesUseCase.call(targets: (1, 3, 6));
```
Comment (mattmarcello) [inline lib/presentation/screens/compose_detach_utxo/bloc/compose_detach_utxo_bloc.dart:59]: we no longer do anything with the targets, should just remove. 


```
class GetFeeEstimatesUseCase {
  final FeeEstimatesRespository feeEstimatesRepository;

  const GetFeeEstimatesUseCase({
    required this.feeEstimatesRepository,
  });

  Future<FeeEstimates> call({required (int, int, int) targets}) async {
    return feeEstimatesRepository.getFeeEstimates().run().then(
          (either) => either.fold(
            (error) => throw Exception("GetFeeEstimates failure"),
            (feeEstimates) => feeEstimates,
          ),
        );
  }
}
```

**Review by mattmarcello — APPROVED** [review]
